src/engine/strat_engine/backstore/devices.rs

```diff
@@ -237,9 +239,32 @@ pub fn initialize_devices(
         dev_info: &DeviceInfo,
         pool_uuid: PoolUuid,
         mda_data_size: MDADataSize,
+        key_description: Option<&str>,
     ) -> StratisResult<StratBlockDev> {
-        let mut f = OpenOptions::new().write(true).open(&dev_info.devnode)?;
+        let dev_uuid = Uuid::new_v4();
+        let (path, deviceno) = match key_description {
+            Some(desc) => {
+                let path = initialize_encrypted_stratis_device(
+                    &dev_info.devnode,
+                    pool_uuid,
+                    dev_uuid,
+                    desc,
+                )?;
+                let dm = DM::new()?;
+                let info = dm.device_info(&DevId::Name(DmName::new(&name_from_uuids(
```

Comment:
The method ```name_from_uuids``` should possibly be defined in ```src/engine/strat_engine/names.rs```.

Reply:
Let's talk about it in stand up. This is only related to crypt devices so I could see it going either way based on what is already in the `names.rs` module.

Reply:
Feel free to leave it here for now.

Reply:
Okay, will do.